agent/consul/state/catalog.go

```diff
@@ -2943,14 +2943,8 @@ func (s *Store) ServiceTopology(
 		maxIdx = idx
 	}
 
-	var (
-		seenUpstreams   = make(map[string]struct{})
```

Comment:
I was trying to refactor upstream and downstream processing but it got too complex for this PR.
`seenUpstreams` and `seenDownstreams` were redundant here and can be safely removed